criu/seize.c

```diff
@@ -535,8 +541,10 @@ static int freeze_processes(void)
 	}
 
 err:
-	if (exit_code == 0 || origin_freezer_state == THAWED)
-		exit_code = freezer_write_state(fd, THAWED);
+	if (exit_code == 0 || origin_freezer_state == THAWED) {
+		ret = freezer_write_state(fd, THAWED);
+		exit_code = exit_code ? exit_code : ret;
```

Comment:
I believe it's enough to return -1 on error and 0 on succes from freeze_processes(). This way code would probably look simpler with something like this:
```
if (freezer_write_state(fd, THAWED))
        exit_code = -1;
```